File: Python language/Searching and Sorting/Binary search.py
def binary_search(list,target):
    l=0
    h=len(list)-1
    while l<=h:
        m=(l+h)//2
        if list[m]<target :
            l=m+1
        elif list[m]==target:
            return m
        else:
            h=m-1
    return None

def verify(index):
    if index is None:
        print("The element is not found")
    else:
        print("The element is found at the index : ",index)

n=int(input("Enter the size of the list : "))
list=[]

for i in range(0,n):
    element=int(input("Enter an element at the index "+str(i)+" : "))
    list.append(element)
print(list)
s=int(input("Enter the element to search : "))
verify(binary_search(list,s))

File: C++ language/Searching/Binary_search.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int n;
    cin>>n;
    int arr[n];

    for(int i=0;i<n;i++)
    cin>>arr[i];

    for(int i=0;i<n;i++)
    {
        for(int j=0;j<n-1;j++)
        {
            int a;
            
            if(arr[j]>arr[i])
            {
                a=arr[i];
                arr[i]=arr[j];
                arr[j]=a;
            }
        }
    }
    cout<<"The sorted array is"<<endl;
    for(int i=0;i<n;i++)
    cout<<arr[i]<<endl;

    int s;
    cout<<"Enter the number to be searched"<<endl;
    cin>>s;

    int low=0,high=n-1,mid;

    while(low<=high)
    {
        mid=(low+high)/2;

        if(arr[mid]<s)
        {
            low=mid+1;
        }
        else if(arr[mid]==s)
        {
            cout<<"The element is found at the index "<<mid<<endl;
            break;
        }
        else
        {
            high=mid-1;
        }
    }

    if(low>high)
    cout<<"The number is not found"<<endl;
}